### PR TITLE
feat: add self-documenting `.env.example` file

### LINKED ISSUE #56468: Add self-documenting `.env.example` file
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
n/a
```

### Which example does this report relate to?

https://github.com/vercel/next.js/tree/canary/packages/create-next-app/templates

### What browser are you using? (if relevant)

_No response_

### How are you deploying your application? (if relevant)

_No response_

### Describe the Bug

It's not very clear to newcomers how `.env.local` files might work with Next.js.

### Expected Behavior

It's very clear how `.env.local` files work with Next.js, and how to get started using one.

### To Reproduce

n/a

### PR DESCRIPTION
**Pull request by hirehamir**
- fixes #56468

### CODE REVIEW

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 874d6e2e71a8450fe52e831c5e6f8a993cac74d6

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

File: packages/create-next-app/templates/app-tw/js/.env.example [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# If you would like to use environment variables for local development,
+# you can copy and paste this file, renaming it to `.env.local`,
+# and add or update any environment variables there.
+YOUR_ENVIRONMENT_VARIABLE="your-value-here"
```
Comment (styfle) [inline packages/create-next-app/templates/app-tw/js/.env.example:4]:
```suggestion
# Rename this file to `.env.local` to use environment variables locally with `next dev`
# https://nextjs.org/docs/app/building-your-application/configuring/environment-variables
MY_HOST="example.com"
```

File: packages/create-next-app/templates/app-tw/ts/.env.example [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# If you would like to use environment variables for local development,
+# you can copy and paste this file, renaming it to `.env.local`,
+# and add or update any environment variables there.
+YOUR_ENVIRONMENT_VARIABLE="your-value-here"
```
Comment (styfle) [inline packages/create-next-app/templates/app-tw/ts/.env.example:4]:
```suggestion
# Rename this file to `.env.local` to use environment variables locally with `next dev`
# https://nextjs.org/docs/app/building-your-application/configuring/environment-variables
MY_HOST="example.com"
```

File: packages/create-next-app/templates/app/js/.env.example [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# If you would like to use environment variables for local development,
+# you can copy and paste this file, renaming it to `.env.local`,
+# and add or update any environment variables there.
+YOUR_ENVIRONMENT_VARIABLE="your-value-here"
```
Comment (styfle) [inline packages/create-next-app/templates/app/js/.env.example:4]:
```suggestion
# Rename this file to `.env.local` to use environment variables locally with `next dev`
# https://nextjs.org/docs/app/building-your-application/configuring/environment-variables
MY_HOST="example.com"
```

File: packages/create-next-app/templates/app/ts/.env.example [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# If you would like to use environment variables for local development,
+# you can copy and paste this file, renaming it to `.env.local`,
+# and add or update any environment variables there.
+YOUR_ENVIRONMENT_VARIABLE="your-value-here"
```
Comment (styfle) [inline packages/create-next-app/templates/app/ts/.env.example:4]:
```suggestion
# Rename this file to `.env.local` to use environment variables locally with `next dev`
# https://nextjs.org/docs/app/building-your-application/configuring/environment-variables
MY_HOST="example.com"
```

File: packages/create-next-app/templates/default-tw/js/.env.example [hunk 5]
```diff
@@ -0,0 +1,4 @@
+# If you would like to use environment variables for local development,
+# you can copy and paste this file, renaming it to `.env.local`,
+# and add or update any environment variables there.
+YOUR_ENVIRONMENT_VARIABLE="your-value-here"
```
Comment (styfle) [inline packages/create-next-app/templates/default-tw/js/.env.example:4]:
```suggestion
# Rename this file to `.env.local` to use environment variables locally with `next dev`
# https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables
MY_HOST="example.com"
```

File: packages/create-next-app/templates/default-tw/ts/.env.example [hunk 6]
```diff
@@ -0,0 +1,4 @@
+# If you would like to use environment variables for local development,
+# you can copy and paste this file, renaming it to `.env.local`,
+# and add or update any environment variables there.
+YOUR_ENVIRONMENT_VARIABLE="your-value-here"
```
Comment (styfle) [inline packages/create-next-app/templates/default-tw/ts/.env.example:4]:
```suggestion
# Rename this file to `.env.local` to use environment variables locally with `next dev`
# https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables
MY_HOST="example.com"
```

File: packages/create-next-app/templates/default/js/.env.example [hunk 7]
```diff
@@ -0,0 +1,4 @@
+# If you would like to use environment variables for local development,
+# you can copy and paste this file, renaming it to `.env.local`,
+# and add or update any environment variables there.
+YOUR_ENVIRONMENT_VARIABLE="your-value-here"
```
Comment (styfle) [inline packages/create-next-app/templates/default/js/.env.example:4]:
```suggestion
# Rename this file to `.env.local` to use environment variables locally with `next dev`
# https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables
MY_HOST="example.com"
```

File: packages/create-next-app/templates/default/ts/.env.example [hunk 8]
```diff
@@ -0,0 +1,4 @@
+# If you would like to use environment variables for local development,
+# you can copy and paste this file, renaming it to `.env.local`,
+# and add or update any environment variables there.
+YOUR_ENVIRONMENT_VARIABLE="your-value-here"
```
Comment (styfle) [inline packages/create-next-app/templates/default/ts/.env.example:4]:
```suggestion
# Rename this file to `.env.local` to use environment variables locally with `next dev`
# https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables
MY_HOST="example.com"
```

**Review by styfle — APPROVED** [review]

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | hamirmahal/next.js feat/add-self-documenting-env-example-file | Change |
| - | - | - | - |
| buildDuration | 10.4s | 10.4s | N/A |
| buildDurationCached | 6.4s | 6.3s | N/A |
| nodeModulesSize | 172 MB | 172 MB | ✓ |
| nextStartRea..uration (ms) | 895ms | 896ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | hamirmahal/next.js feat/add-self-documenting-env-example-file | Change |
| - | - | - | - |
| 199-HASH.js gzip | 27.5 kB | 27.5 kB | N/A |
| 3f784ff6-HASH.js gzip | 50.9 kB | 50.9 kB | ✓ |
| 99.HASH.js gzip | 182 B | 182 B | ✓ |
| framework-HASH.js gzip | 45.3 kB | 45.3 kB | ✓ |
| main-app-HASH.js gzip | 254 B | 252 B | N/A |
| main-HASH.js gzip | 32.9 kB | 32.9 kB | N/A |
| webpack-HASH.js gzip | 1.75 kB | 1.75 kB | N/A |
| Overall change | 96.4 kB | 96.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | hamirmahal/next.js feat/add-self-documenting-env-example-file | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | hamirmahal/next.js feat/add-self-documenting-env-example-file | Change |
| - | - | - | - |
| _app-HASH.js gzip | 206 B | 205 B | N/A |
| _error-HASH.js gzip | 182 B | 180 B | N/A |
| amp-HASH.js gzip | 506 B | 505 B | N/A |
| css-HASH.js gzip | 322 B | 323 B | N/A |
| dynamic-HASH.js gzip | 2.57 kB | 2.57 kB | N/A |
| edge-ssr-HASH.js gzip | 260 B | 259 B | N/A |
| head-HASH.js gzip | 350 B | 350 B | ✓ |
| hooks-HASH.js gzip | 369 B | 369 B | ✓ |
| image-HASH.js gzip | 4.35 kB | 4.35 kB | N/A |
| index-HASH.js gzip | 256 B | 256 B | ✓ |
| link-HASH.js gzip | 2.64 kB | 2.63 kB | N/A |
| routerDirect..HASH.js gzip | 312 B | 311 B | N/A |
| script-HASH.js gzip | 385 B | 384 B | N/A |
| withRouter-HASH.js gzip | 307 B | 308 B | N/A |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 1.08 kB | 1.08 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | hamirmahal/next.js feat/add-self-documenting-env-example-file | Change |
| - | - | - | - |
| _buildManifest.js gzip | 485 B | 482 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | hamirmahal/next.js feat/add-self-documenting-env-example-file | Change |
| - | - | - | - |
| index.html gzip | 528 B | 530 B | N/A |
| link.html gzip | 540 B | 543 B | N/A |
| withRouter.html gzip | 524 B | 526 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | hamirmahal/next.js feat/add-self-documenting-env-example-file | Change |
| - | - | - | - |
| edge-ssr.js gzip | 93.8 kB | 93.8 kB | ✓ |
| page.js gzip | 152 kB | 152 kB | N/A |
| Overall change | 93.8 kB | 93.8 kB | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | hamirmahal/next.js feat/add-self-documenting-env-example-file | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 624 B | 624 B | ✓ |
| middleware-r..fest.js gzip | 150 B | 151 B | N/A |
| middleware.js gzip | 22.9 kB | 22.9 kB | N/A |
| edge-runtime..pack.js gzip | 1.92 kB | 1.92 kB | ✓ |
| Overall change | 2.55 kB | 2.55 kB | ✓ |

</details>

</details>
Commit: b0884e929074ba4e9c4c57031a72779ed0333da3

**Pull request merged**